meinberlin/apps/plans/models.py

```diff
@@ -208,6 +208,28 @@ def published_projects(self):
             is_archived=False,
         )
 
+    @cached_property
+    def published_projects_count(self) -> int:
+        try:
+            prefetch_enabled = (
+                self.projects.prefetch_cache_name in self._prefetched_objects_cache
+            )
+        except AttributeError:
+            prefetch_enabled = False
+
+        if prefetch_enabled:
+            count = 0
+            for project in self.projects.all():
+                if project.is_draft or project.is_archived:
+                    continue
+                if project.access in [Access.PUBLIC, Access.SEMIPUBLIC]:
+                    count += 1
+
+            return count
+
+        else:
+            return self.published_projects.count()
+
```

Comment:
why not move this to the plan serializer method? we can't use the cached_property decorator, but we won't need to check for the prefetch cache name.

Reply:
Yeah, I don't think the `cached_property` is important for this story. I just added it because why not.

> why not move this to the plan serializer method?

Do you mean move the whole function to the serializer? If so, my thinking was that there may be other users of this code (outside of serializing). Like the property `published_projects` just above it is also defined for the model.

But I am pretty indifferent here. Shall I go ahead and move it to the serializer?

Reply:
decision see next post